packages/manager/modules/product-offers/src/product-offers.controller.js

```diff
@@ -0,0 +1,56 @@
+import OrderWorkflow from './workflows/product-offers-order-workflow.class';
+import ServicesWorkflow from './workflows/product-offers-services-workflow.class';
+
+import DetachService from './services/product-offers-detach.service';
+
+import { WORKFLOW_TYPES } from './product-offers.constants';
+
+export default class ProductOffersController {
+  /* @ngInject */
+  constructor($q, $translate, WucOrderCartService) {
+    this.$q = $q;
+    this.$translate = $translate;
+    this.WucOrderCartService = WucOrderCartService;
+  }
+
+  $onInit() {
+    if (!this.workflowType) {
+      throw new Error('ovhProductOffersComponent: workflowType is undefined');
+    }
+
+    if (!this.workflowOptions) {
+      throw new Error(
+        'ovhProductOffersComponent: workflowOptions is undefined',
+      );
+    }
+
+    switch (this.workflowType) {
+      case WORKFLOW_TYPES.ORDER:
+        this.workflow = new OrderWorkflow(
+          this.$q,
+          this.$translate,
+          this.workflowOptions,
+          this.WucOrderCartService,
+        );
+        break;
+      case WORKFLOW_TYPES.SERVICES:
+        this.workflow = new ServicesWorkflow(
```

Comment:
you can reuse creating ServicesWorkflow object code by defining it in a separate function like createServicesWorkflow()

Reply:
Hmm why not but the workflows look different here, `ServicesWorkflow` is used once.